pkg/fleet-manager/fleet_plugin.go

```diff
@@ -35,29 +37,68 @@ const (
 	PrometheusThanosServiceName = "prometheus-prometheus-thanos"
 
 	NoneClusterIP = "None"
+
+	FleetPluinLabel = "fleet.kurator.dev/fleet-plugin"
 )
 
 func (f *FleetManager) reconcilePlugins(ctx context.Context, fleet *fleetapi.Fleet, fleetClusters map[ClusterKey]*fleetCluster) (ctrl.Result, error) {
 	var resources kube.ResourceList
 
 	if fleet.Spec.Plugin != nil {
-		result, ctrlResult, err := f.reconcileMetricPlugin(ctx, fleet, fleetClusters)
-		if err != nil || ctrlResult.RequeueAfter > 0 {
-			return ctrlResult, err
+		type reconcileResult struct {
+			result     kube.ResourceList
+			ctrlResult ctrl.Result
+			err        error
+		}
+
+		type reconcileFunc func(context.Context, *fleetapi.Fleet, map[ClusterKey]*fleetCluster) (kube.ResourceList, ctrl.Result, error)
+
+		funcs := []reconcileFunc{
+			f.reconcileMetricPlugin,
+			f.reconcileGrafanaPlugin,
+			f.reconcileKyvernoPlugin,
+			f.reconcileBackupPlugin,
+		}
+
+		resultsChannel := make(chan reconcileResult, len(funcs))
+		var wg sync.WaitGroup
+
+		for _, fn := range funcs {
+			wg.Add(1)
+			go func(fn reconcileFunc) {
+				defer wg.Done()
+				result, ctrlResult, err := fn(ctx, fleet, fleetClusters)
+				resultsChannel <- reconcileResult{result, ctrlResult, err}
+			}(fn)
+		}
+
+		go func() {
+			wg.Wait()
+			close(resultsChannel)
+		}()
+
+		var errs []error
+		var ctrlResults []ctrl.Result
+
+		for res := range resultsChannel {
+			if res.err != nil {
+				errs = append(errs, res.err)
+			}
+			if res.ctrlResult.RequeueAfter > 0 {
+				ctrlResults = append(ctrlResults, res.ctrlResult)
+			}
+			resources = append(resources, res.result...)
 		}
-		resources = append(resources, result...)
 
-		result, ctrlResult, err = f.reconcileGrafanaPlugin(ctx, fleet)
-		if err != nil || ctrlResult.RequeueAfter > 0 {
-			return ctrlResult, err
+		if len(errs) > 0 {
+			// Combine all errors into one error message
+			return ctrl.Result{}, fmt.Errorf("encountered multiple errors: %v", errs)
 		}
-		resources = append(resources, result...)
 
-		result, ctrlResult, err = f.reconcileKyvernoPlugin(ctx, fleet, fleetClusters)
-		if err != nil || ctrlResult.RequeueAfter > 0 {
-			return ctrlResult, err
+		if len(ctrlResults) > 0 {
+			// Handle multiple ctrlResults
+			return ctrlResults[0], nil // TODO: if we need strategy about RequeueAfter
```

Comment:
Yes, i think this is not right.

If there are > 1 plugins, we only check the first one. If the 1st installed successfully, but the 2nd is being installed, we may wrongly treat all plugins ready,